src/main/java/reactor/core/publisher/BlockingSink.java

```diff
@@ -278,14 +280,11 @@ public Emission finish() {
 		}
 	}
 
-	@Override
 	public long getCapacity() {
-		return Trackable.class.isAssignableFrom(actual.getClass()) ? ((Trackable)
-				actual).getCapacity() :
-				Long.MAX_VALUE;
+		return Scannable.from(actual)
+		                .scanOrDefault(Scannable.Attr.CAPACITY, Integer.MAX_VALUE);
```

Comment:
should the change already be made in 3.0.6 though? would have preferred to have a separate change for that, closing issue 397, or even defer it to 3.1.0 as initially planned...